pkg/controller/registry/resolver/resolver.go

```diff
@@ -184,9 +183,32 @@ func (resolver *MultiSourceResolver) resolveCRDDescription(sourceRefs []registry
 		return v1alpha1.StepResource{}, "", fmt.Errorf("Unknown CRD %s", crdKey)
 	}
 
-	// TODO: Change to lookup the CSV from the preferred or default channel.
-	logger.WithField("owner", csvs[0].CSV.Name).Info("found owner")
-	return v1alpha1.StepResource{}, csvs[0].CSV.Name, nil
+	var ownerName string
+	if owners, ok := existingCRDOwners[crdKey.Name]; ok && len(owners) > 0 {
```

Comment:
could we make this control flow clearer?

doesn't have to be this, but something that removes the nesting:
```go
owners, ok := existingCRDOwners[crdKey.Name]
if !ok {
  ownerCount = 0 
} else {
  ownerCount = len(owners)
}
switch len(owners):
  case 0:
  // get default channel
  case 1:
  // found an owner
  case >1:
  // error
}
```